src/App.tsx

```diff
@@ -1,39 +1,139 @@
+import React, { useState } from 'react';
 import './App.scss';
 import Carousel from './components/Carousel';
 
-interface State {
-  images: string[];
-}
-
-class App extends React.Component<{}, State> {
-  state = {
-    images: [
-      './img/1.png',
-      './img/2.png',
-      './img/3.png',
-      './img/4.png',
-      './img/5.png',
-      './img/6.png',
-      './img/7.png',
-      './img/8.png',
-      './img/9.png',
-      './img/10.png',
-    ],
-  };
+type StateProps = {
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinite: boolean,
+  imageGap: number,
+};
+
+const App: React.FC = () => {
+  const [state, setState] = useState<StateProps>({
+    step: 3,
+    frameSize: 3,
+    itemWidth: 130,
+    animationDuration: 1000,
+    infinite: false,
+    imageGap: 10,
+  } as StateProps);
+
+  const [pictures] = useState([
+    './img/1.png',
+    './img/2.png',
+    'https://media.glamourmagazine.co.uk/photos/64b6ae8fcd1ad7c51ecf045c/1:1/'
+    + 'w_1280,h_1280,c_limit/RYAN%20GOSLING%20EVA%20MENDES%20180723%20default'
+    + 'GettyImages-1527942629.jpg',
+    './img/3.png',
+    './img/4.png',
+    './img/5.png',
+    './img/6.png',
+    './img/7.png',
+    './img/8.png',
+    './img/9.png',
+    './img/10.png',
+  ]);
 
-  render() {
-    const { images } = this.state;
+  document.documentElement.style.setProperty('--itemLength', `${pictures.length}px`);
+  document.documentElement.style.setProperty('--carouselWidth', `${pictures.length * (state.itemWidth + state.imageGap)}px`);
+  document.documentElement.style.setProperty('--containerWidth', `${(state.itemWidth + state.imageGap) * (state.frameSize) - state.imageGap}px`);
+  document.documentElement.style.setProperty('--animationDuration', `${state.animationDuration}ms`);
+  document.documentElement.style.setProperty('--imageGap', `${state.imageGap}px`);
+
+  const changeHadler = (value: number | boolean,
+    key: keyof StateProps) => {
+    setState(prevState => ({ ...prevState, [key]: value }));
+  };
 
-    return (
-      <div className="App">
-        {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+  return (
+    <div className="App">
+      <h1 data-cy="title">{`Carousel with ${pictures.length} images, ${state.frameSize} frameSize and ${state.step} stepSize`}</h1>
 
-        <Carousel />
-      </div>
-    );
-  }
-}
+      <Carousel
+        images={pictures}
+        step={state.step}
+        itemWidth={state.itemWidth}
+        frameSize={state.frameSize}
+        imageGap={state.imageGap}
+      />
+      <input
+        type="number"
+        name="itemWidth"
+        id="itemWidth"
+        value={state.itemWidth}
+        min={50}
+        max={500}
+        step={10}
+        onChange={e => {
+          changeHadler(+e.target.value, e.target.name as keyof StateProps);
+          document.documentElement.style.setProperty('--itemWidth', `${+e.target.value}px`);
+        }}
```

Comment:
```suggestion
        onChange={changeHadler}
```
in case you have `changeHadler` - handle event there (get name and value of target)
In this case, JSX will look much clearer